engine/src/main/java/org/hibernate/search/engine/impl/ConfigContext.java

```diff
@@ -323,23 +327,46 @@ private void addFilterDef(FullTextFilterDef defAnn) {
 		return Collections.unmodifiableMap( initializedAnalyzers );
 	}
 
-	private void initRemoteAnalyzerReference(Map<String, AnalyzerReference> initializedAnalyzerReferences, LazyAnalyzerReference namedAnalyzerReference, IndexManagerHolder indexesFactory) {
+	private void initLazyRemoteAnalyzer(Map<String, AnalyzerReference> initializedAnalyzers, RemoteAnalyzerReference lazyRemoteAnalyzerReference,
+			IndexManagerHolder indexesFactory) {
+		LazyRemoteAnalyzer lazyAnalyzer = (LazyRemoteAnalyzer) lazyRemoteAnalyzerReference.getAnalyzer();
+
+		if ( initializedAnalyzers.containsKey( lazyAnalyzer.getName() ) ) {
+			AnalyzerReference analyzerReference = initializedAnalyzers.get( lazyAnalyzer.getName() );
+			if ( !( analyzerReference instanceof RemoteAnalyzerReference ) ) {
```

Comment:
No need to change it now but I was wondering if we need add this check.
Wouldn't it be possible to have a definition that applied both locally and remotely? For the simple case for example, like whitespace, simple, and so on